File: chatwoot/csharp/src/OSEG.ChatwootExamples/UpdateAgentBotExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Com.Chatwoot.Api;
using Com.Chatwoot.Client;
using Com.Chatwoot.Model;

namespace OSEG.ChatwootExamples;

public class UpdateAgentBotExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("api_access_token", "USER_API_KEY");
        // config.ApiKey.Add("api_access_token", "AGENT_BOT_API_KEY");
        // config.ApiKey.Add("api_access_token", "PLATFORM_APP_API_KEY");

        var updateAgentBotRequest = new UpdateAgentBotRequest(
            agentBot: 0
        );

        try
        {
            new InboxesApi(config).UpdateAgentBot(
                accountId: 0,
                id: 0,
                data: updateAgentBotRequest
            );
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling InboxesApi#UpdateAgentBot: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: chatwoot/typescript-axios/src/UpdateAgentBotExample.ts
import axios, { AxiosError } from "axios";
import * as api from "chatwoot_client"

const configuration = new api.Configuration({
  apiKey: "USER_API_KEY",
  // apiKey: "AGENT_BOT_API_KEY",
  // apiKey: "PLATFORM_APP_API_KEY",
});

const updateAgentBotRequest: api.UpdateAgentBotRequest = {
  agent_bot: 0,
};

new api.InboxesApi(configuration).updateAgentBot(
  0, // accountId
  0, // id
  updateAgentBotRequest, // data
).catch((error: Error | AxiosError) => {
  console.log("Exception when calling InboxesApi#updateAgentBot:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
